src/gui/render.cpp

```diff
@@ -615,26 +618,35 @@ void RENDER_SetForceUpdate(bool f) {
 }
 
 #if C_OPENGL
-static bool RENDER_GetShader(std::string& shader_path, char *old_src) {
+static bool RENDER_GetShader(std::string &shader_path, char *old_src)
+{
+	const std::unordered_map<std::string, const char *> builtin_shaders = {
+	        {"advinterp2x", advinterp2x_glsl},
+	        {"advinterp3x", advinterp3x_glsl},
+	        {"advmame2x", advmame2x_glsl},
+	        {"advmame3x", advmame3x_glsl},
+	        {"crt-easymode-flat", crt_easymode_tweaked_glsl},
+	        {"crt-fakelottes-flat", crt_fakelottes_tweaked_glsl},
+	        {"rgb2x", rgb2x_glsl},
+	        {"rgb3x", rgb3x_glsl},
+	        {"scan2x", scan2x_glsl},
+	        {"scan3x", scan3x_glsl},
+	        {"sharp", sharp_glsl},
+	        {"tv2x", tv2x_glsl},
+	        {"tv3x", tv3x_glsl},
```

Comment:
Almost time we can retire `scalar = <software implementation>` for good!

Reply:
There's a biiig problem with retiring them :( Scalars are exposed via our CLI interface, and consequently they are being used by GUI frontends this way (e.g. Lutris). Also, glshaders for some of those legacy options seem to be sub-par.

Reply:
Yes; we need to never break the user experience or external tools. 

Perhaps `scalar` (and command line argument) could map over to the equivalent `glshader`, and even try opengl output. If that doesn't work we could fallback to texture mode, and then recommend the user get opengl working on their device.

Reply:
> glshaders for some of those legacy options seem to be sub-par.

That's a more serious problem.. I suppose they're just as bad even with openglnb or pp?

Reply:
Hmm, I don't remember if I tested those older glshaders with openglnb… but if it matters, we should make it switch automatically, so user doesn't need to.